server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java

```diff
@@ -1236,7 +1252,24 @@ public void flushOneMemTable() {
                 tsFileResource.getTsFile().getName(),
                 e1);
           }
-          Thread.currentThread().interrupt();
+          // release resource
+          try {
+            syncReleaseFlushedMemTable(memTableToFlush);
+            for (CloseFileListener closeFileListener : closeFileListeners) {
+              closeFileListener.onClosed(this);
```

Comment:
This will register this file as SealedTsFile into TsFileManager, which may cause bugs in query

Reply:
Fixed, I will set time index to its initial status.